testing/sdk_metrics.py

```diff
@@ -24,6 +25,22 @@ def get_scheduler_metrics(service_name, timeout_seconds=15 * 60):
     return sdk_cmd.service_request("GET", service_name, "/v1/metrics").json()
 
 
+def get_failure_metrics(service_name: str) -> typing.Dict:
+    history = sdk_cmd.history_request().json()
+    service_history = [h for h in history["frameworks"] if h["name"] == service_name]
+    if not service_history:
+        return dict()
+
+    def collect():
+        failure_keys = ["TASK_FAILED", "TASK_ERROR"]
+        for k, v in service_history[0].items():
+            if k not in failure_keys:
+                continue
+            yield k, v
+
+    return dict(collect())
```

Comment:
Since we're never using the individual values for the failure modes, could we not just return the `sum()` of the counts here? This would make its usage at the call-sites much simpler.

Reply:
If at some point we need the additional functionality, we can refactor this (maybe for #2638), but I would say that we should look for the simplest solution here.